core/templates/pages/skill-editor-page/editor-tab/skill-editor-main-tab.directive.ts

```diff
@@ -116,6 +116,9 @@ angular.module('oppia').directive('skillEditorMainTab', [
 
           $scope.hasLoadedSkill = function() {
             $scope.skill = SkillEditorStateService.getSkill();
+            // TODO(#8521): Remove the use of $rootScope.$applyAsync()
+            // once the directive is migrated to angular.
+            $rootScope.$applyAsync();
             return SkillEditorStateService.hasLoadedSkill();
```

Comment:
This is needed because the change count in the navbar is not getting updated, as change detection is not working here due to hybrid mode. So we have to apply it manually.

Reply:
We should be careful when applying change detection manually. We should not be calling triggering a digest loop on methods that are directly used in the template (e.g. ngIf) -- doing so ends up calling the digest loop more often than actually required. We could instead trigger it when there is some manipulation of data (which is often the case when we need the view to update). For example, we could trigger a digest loop when there is a new change added to the change list or similar.

Reply:
Got it, will take care of this next time. 
Thanks, @kevintab95.

Reply:
Is it fine now (added it in ng-change)?

Reply:
I'm a bit confused, how does "getAssignedSkillTopicData" method relate to the action of the "change count" updating? 

> This is needed because the change count in the navbar is not getting updated, as change detection is not working here due to hybrid mode. So we have to apply it manually.

Can we figure out exactly when we need the change count in the navbar to update and then trigger change detection there?

Reply:
I tried to resolve the issue by emitting an event everything a misconception is added/edited/deleted and updating the view in the parent component (skill-editor-main-tab.directive.ts ) every time, which saves us from running an infinite loop of change detection as now the change detection is applied only when either we are adding, deleting or editing the misconception.
Proofs:

https://user-images.githubusercontent.com/86483059/149196484-f07e6a56-803d-4bdd-8f74-f89f39fca1d9.mp4

Reply:
Sounds good, thanks!